java-frontend/src/main/java/org/sonar/java/se/ExplodedGraphWalker.java

```diff
@@ -561,13 +562,27 @@ private void visit(Tree tree, @Nullable Tree terminator) {
       case METHOD_REFERENCE:
         programState = programState.stackValue(constraintManager.createSymbolicValue(tree));
         break;
+      case ASSERT_STATEMENT:
+        executeAssertStatement(tree);
+        return;
       default:
     }
 
     checkerDispatcher.executeCheckPostStatement(tree);
     clearStack(tree);
   }
 
+  private void executeAssertStatement(Tree tree) {
+    // After an assert statement we know that the evaluated expression is true.
+    ProgramState.Pop pop = programState.unstackValue(1);
+    pop.values.forEach(v -> v.setConstraint(pop.state, BooleanConstraint.TRUE)
+      .forEach(ps -> {
+        checkerDispatcher.syntaxNode = tree;
```

Comment:
If it would be then it would be propagated through method calls within the class => field